internal/cli/root.go

```
@@ -7,6 +7,78 @@ import (
+// WithAncestorFlags appends reg to the ctx-carried slice of ancestor flag
+// registrars and returns the new context. Groups call this during Run so
+// child commands inherit the Group's declared flags; the slice preserves
+// registration order (outermost ancestor first) so leaf tests can reason
+// about precedence by inserting guards.
+//
+// Per stdlib context.WithValue contract, ctx must not be nil.
+func WithAncestorFlags(ctx context.Context, reg func(*flag.FlagSet)) context.Context {
+	prior, _ := ctx.Value(ancestorFlagsKey{}).([]flagRegistrar)
+	next := make([]flagRegistrar, 0, len(prior)+1)
+	next = append(next, prior...)
+	next = append(next, flagRegistrar(reg))
```

golangci-lint (2.11.4)

[error] unnecessary conversion

(unconvert)